captcha-plugin-draft.txt

```diff
@@ -107,6 +107,13 @@ We provisionally instantiate these with
 
 [TODO: I would really like to use something other than RSA, but haven't found
 satisfactory research on alternative blind signature schemes.]
+[Alex: It could be possible to use a DSA-style blind signature (see
+http://blog.cryptographyengineering.com/p/note-on-blind-signature-schemes.html).
+The protocol requires three steps, the additional step is required since the
+server initially sends a random element for each signature that needs to be
+blindly signed. While this is not ideal it may be possible to incorporate
+this into the structure below so that the plugin asks for 10000 elements of
+randomness from the server when it generates the nonces.]
```

Comment:
An extra round trip might cost more than RSA costs.  Also, RSA blind signing is already information theoretically secure for the user, so the Tor project should probably reject any scheme that fails to be information theoretically secure. 

In Taler, I just added a full domain hash that employs the denomination key as a seed.  I think that's as close to RSA-PSS as one can get.  And it's provably secure for the mint assuming the RSA-KTI assumption.  I'll add the notes to our paper soonish. 

We only need a 1024 bit RSA key here since the key should rotate with every TBB release.  Arguably a 768 bit key suffices, certainly if we rotate the key every couple months using the auditor arrangement in Taler.